### PR TITLE
LPS-167150 Migrate Experience Selector in view mode to Picker component

### PR DESCRIPTION
**Pull request by claraizquierdo**
**This PR depends on** [https://github.com/liferay/clay/issues/5271 .](https://github.com/liferay-frontend/liferay-portal/pull/2919) .

Motivation
=======
The Screen Reader is not announcing the exact role and state for the select experience combo box.
[Link to ticket](https://issues.liferay.com/browse/LPS-167150)
This PR is also solves another issue: The combo box needs an informative label to display information about what it is.
[Link to ticket](https://issues.liferay.com/browse/LPS-167158)

Proposed Solution
========
Migrate the experience selector to the new Picker component https://clayui.com/docs/components/picker.html. This new component add the role correctly. We need to add title and aria-label attributes so the pop-up label with the information is displayed.

Steps to Verify:
----------------

1. Go to Top Menu > Edit
2. Navigate to Default Experience Combo Box
3. Create and publish a New Experience
4. Navigate to Default Combo box
5. Listen to Screen Reader (first story)
6. Hover on the Combo box -> a pop-up with the message 'Experience Selector' should appear.

Screenshots (optional):
-----------------------

### CODE REVIEW

**Comment by claraizquierdo** [comment]
Hi,

there is some work left to be done in this PR:
- @cgoncas @Mikellorza could you help us with how the properties are passed to the new React Component in `modules/apps/segments/segments-web/src/main/resources/META-INF/resources/segments_experience_selector.jsp` file please?
- @Stephy6 the ExperiencePicker component does not have any tests. I'm not sure if they are necessary since almost all the code of the component is a Clay component (that should be tested in Clay). Feel free to add some tests if you think they are necessary. 
- I've seen a weir behavior with he VoiceOver screen reader. I've asked here for advice: https://liferay.slack.com/archives/C04CPMRMXAL/p1672242567217899

Anyway, this code is working and those unfinished tasks could be addressed in another PR if you prefer.

**Comment by cgoncas** [comment]
ci:test:sf

**Comment by cgoncas** [comment]
ci:test:relevant

**Comment by cgoncas** [comment]
ci:test:relevant

**Comment by cgoncas** [comment]
ci:test:relevant

**Comment by cgoncas** [comment]
ci:test:sf

**Comment by cgoncas** [comment]
ci:forward

**Comment by cgoncas** [comment]
ci:test:sf

**Comment by cgoncas** [comment]
ci:forward

**Comment by cgoncas** [comment]
ci:test:relevant

**Comment by cgoncas** [comment]
@CarlosBrichete

File: modules/apps/segments/segments-web/src/main/java/com/liferay/segments/web/internal/display/context/SegmentsExperienceSelectorDisplayContext.java [hunk 1]
```diff
@@ -71,48 +73,12 @@ public SegmentsExperienceSelectorDisplayContext(
...
-		for (SegmentsExperience segmentsExperience : segmentsExperiences) {
-			segmentsExperiencesJSONArray.put(
-				_getSegmentsExperienceJSONObject(
-					segmentsExperience, segmentsExperiences));
-		}
-
-		return segmentsExperiencesJSONArray;
+	public Map<String, Object> getData() throws PortalException {
+		return HashMapBuilder.<String, Object>put(
+			"experiences", _getSegmentsExperiencesJSONArray()
```
Comment (cgoncas) [inline modules/apps/segments/segments-web/src/main/java/com/liferay/segments/web/internal/display/context/SegmentsExperienceSelectorDisplayContext.java:78]: the JSON should be segmentsExperiences

File: modules/apps/segments/segments-web/src/main/java/com/liferay/segments/web/internal/display/context/SegmentsExperienceSelectorDisplayContext.java [hunk 2]
```diff
@@ -71,48 +73,12 @@ public SegmentsExperienceSelectorDisplayContext(
...
+		).put(
+			"selectedExperience", _getSegmentsExperienceSelectedJSONObject()
```
Comment (cgoncas) [inline modules/apps/segments/segments-web/src/main/java/com/liferay/segments/web/internal/display/context/SegmentsExperienceSelectorDisplayContext.java:80]: the JSON should be selectedSegmentsExperiences

Comment (claraizquierdo) [reply modules/apps/segments/segments-web/src/main/java/com/liferay/segments/web/internal/display/context/SegmentsExperienceSelectorDisplayContext.java:80]: @cgoncas , shouldn't it be in singular? `selectedSegmentsExperience` 🤔

Comment (cgoncas) [reply modules/apps/segments/segments-web/src/main/java/com/liferay/segments/web/internal/display/context/SegmentsExperienceSelectorDisplayContext.java:80]: Yes it is :)

Comment (claraizquierdo) [reply modules/apps/segments/segments-web/src/main/java/com/liferay/segments/web/internal/display/context/SegmentsExperienceSelectorDisplayContext.java]: Changed!

**Comment by cgoncas** [comment]
Hey @claraizquierdo and @Stephy6 !!

@Stephy6 (https://issues.liferay.com/browse/LPS-167158) changes is included in this PR?

Thanks!!

**Comment by claraizquierdo** [comment]
> Hey @claraizquierdo and @Stephy6 !!
> 
> @Stephy6 (https://issues.liferay.com/browse/LPS-167158) changes is included in this PR?
> 
> Thanks!!

Not yet @cgoncas because we have been discussing the best solution for the bug. The best solution implies adding a visible <label> before the select, and for that we will need design changes. For now, we can add an aria-label to the select and continue working on the best solution in another pull. What do you think?

**Comment by Stephy6** [comment]
> Hey @claraizquierdo and @Stephy6 !!
> 
> @Stephy6 (https://issues.liferay.com/browse/LPS-167158) changes is included in this PR?
> 
> Thanks!!

Now it does @cgoncas.

**Comment by claraizquierdo** [comment]
ci:test:sf

**Comment by claraizquierdo** [comment]
ci:test:relevant

**Comment by Stephy6** [comment]
ci:test:sf

**Comment by cgoncas** [comment]
ci:test:relevant

**Comment by claraizquierdo** [comment]
ci:test:stable

**Comment by claraizquierdo** [comment]
ci:test:relevant

**Comment by claraizquierdo** [comment]
ci:test:relevant

**Comment by claraizquierdo** [comment]
Hi @CarlosBrichete and @rafaellajordao ,

some tests that interact with the experience selector in view mode need to be updated. I need to send this pull asap, do you think I can send this PR directly to Brian with these tests failing?

**Comment by CarlosBrichete** [comment]
Hey @claraizquierdo, there 3 tests failing just in this PR:
* ExperienceCreateExperience#AddNonInstantiableWidget
* ExperienceCreateExperience#MapMultipleFragmentsBetweenExperiences
* ExperienceCreateExperience#MapFragmentsMultipleExperiences

**Comment by claraizquierdo** [comment]
> Hey @claraizquierdo, there 3 tests failing just in this PR:
> 
> * ExperienceCreateExperience#AddNonInstantiableWidget
> * ExperienceCreateExperience#MapMultipleFragmentsBetweenExperiences
> * ExperienceCreateExperience#MapFragmentsMultipleExperiences

And do you think that they fail because of the changes in the experience selector html or is there something else?

**Comment by CarlosBrichete** [comment]
ci:test:relevant

**Comment by CarlosBrichete** [comment]
ci:test:bundle

**Comment by CarlosBrichete** [comment]
ci:test:relevant

**Comment by CarlosBrichete** [comment]
ci:reevaluate:1397401_2302

**Comment by CarlosBrichete** [comment]
ci:reevaluate:1397401_2302

**Comment by claraizquierdo** [comment]
ci:test:relevant

**Comment by claraizquierdo** [comment]
ci:forward

**Comment by cgoncas** [comment]
> ci:forward

Finally!!! @claraizquierdo.

**Pull request closed without merging**